--- git-svn	2014-12-19 18:08:52.000000000 +0200
+++ git-svn.new	2015-01-07 10:54:02.825521787 +0200
"
" From:
" http://stackoverflow.com/questions/4238876/git-svn-fails-to-dcommit-even-after-clean-checkout
"
@@ -21,6 +21,7 @@
 use Getopt::Long qw/:config gnu_getopt no_ignore_case auto_abbrev/;
 use IPC::Open3;
 use Memoize;
+use Time::HiRes qw/usleep/;
 
 use Git::SVN;
 use Git::SVN::Editor;
@@ -1014,7 +1015,17 @@
 				$gs->{inject_parents_dcommit}->{$cmt_rev} =
 				                               $parents->{$d};
 			}
-			$_fetch_all ? $gs->fetch_all : $gs->fetch;
+                        my $retry;
+                        fetch: for ($retry = 0; $retry < 30; ++$retry) {
+                            $_fetch_all ? $gs->fetch_all : $gs->fetch;
+                            last fetch if ($gs->rev_map_max >= $cmt_rev);
+                            # Asynchronous commit push not complete
+                            usleep(20000 * ($retry + 1));
+                        }
+                        if ($retry > 0 && $gs->rev_map_max < $cmt_rev) {
+                            fatal "New revision $cmt_rev did not appear",
+                                  "in repository after $retry retries.";
+                        }
 			$SVN::Error::handler = $err_handler;
 			$last_rev = $cmt_rev;
 			next if $_no_rebase;
